src/plopp/core/limits.py

```diff
@@ -1,39 +1,64 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
 
-from typing import Literal, Tuple
+from typing import Literal, Tuple, Union
 
 import numpy as np
-from scipp import Variable, scalar
+from scipp import DataArray, Variable, scalar, stddevs
+
+from .utils import merge_masks
 
 
 def find_limits(
-    x: Variable, scale: Literal['linear', 'log'] = 'linear'
+    x: Union[Variable, DataArray], scale: Literal['linear', 'log'] = 'linear'
 ) -> Tuple[Variable, ...]:
     """
     Find sensible limits, depending on linear or log scale.
     If there are no finite values in the array, raise an error.
     If there are no positive values in the array, and the scale is log, fall back to
     some sensible default values.
     """
-    v = x.values
-    finite_inds = np.isfinite(v)
-    if np.sum(finite_inds) == 0:
+    m = np.ones_like(x.values, dtype='bool')
+    if isinstance(x, DataArray) and len(x.masks):
+        m &= ~merge_masks(x.masks).values
+    if not np.any(m):
+        m = np.ones_like(x.values, dtype='bool')
+
+    v = x.values[m]
+    if x.ndim == 1:
+        # in case the plot is 1d, take the error bars into consideration
+        s = (
+            stddevs(x).values
+            if x.variances is not None
+            else np.zeros_like(x.values, dtype='bool')
+        )[m]
+        v = np.concatenate((v, v - s, v + s))
+
+    v = v[np.isfinite(v)]
+    if len(v) == 0:
         raise ValueError("No finite values were found in array. Cannot compute limits.")
-    finite_vals = v[finite_inds]
     finite_max = None
     if scale == "log":
-        positives = finite_vals > 0
-        if np.sum(positives) == 0:
+        if not np.any(v > 0):
             finite_min = 0.1
             finite_max = 1.0
         else:
             initial = (np.finfo if v.dtype.kind == 'f' else np.iinfo)(v.dtype).max
-            finite_min = np.amin(finite_vals, initial=initial, where=finite_vals > 0)
+            finite_min = np.amin(v, initial=initial, where=v > 0)
     else:
-        finite_min = np.amin(finite_vals)
+        finite_min = np.amin(v)
     if finite_max is None:
-        finite_max = np.amax(finite_vals)
+        finite_max = np.amax(v)
+    # Padding only for 1d plots
+    if x.ndim == 1:
```

Comment:
For the same reason as above (data can be 1d for 3d plots), I think whether we apply padding or not should be decided by an argument to the function.